src/interactive-window/editor-integration/codeGenerator.ts

```diff
@@ -76,6 +79,20 @@ export class CodeGenerator implements IInteractiveWindowCodeGenerator {
             lines.forEach((line, index) => (lines[index] = uncommentMagicCommands(line)));
         }
 
+        // If we have a cell magic then don't send the cell marker.
+        // If we do, then don't sen the cell marker, e.g. in the case of `latex` magic it cannot handle python style comments in the cell.
+        // I.e. %%latex magic cannot handle lines that start with the text `#` which exists in cell markers such as `# %%`.
+        if (
+            metadata.interactiveWindowCellMarker &&
+            lines.length &&
+            lines[0].trim() === metadata.interactiveWindowCellMarker.trim()
+        ) {
+            const nonEmptyLines = lines.slice(1).filter((line) => line.trim().length > 0);
+            if (nonEmptyLines.length > 0 && nonEmptyLines[0].trim().startsWith('%%')) {
+                return { lines: lines.slice(1), executableLines: lines.slice(1) };
```

Comment:
Shouldn't this be nonEmptyLines? Otherwise how do you know to slice the first line?

Reply:
I didn't want to strip out the white space, I guess I can do that as well.